awx/api/serializers.py

```diff
@@ -1719,6 +1719,8 @@ def validate_name(self, value):
         name = force_text(value or '')
         # Validate here only, update in main validate method.
         host, port = self._get_host_port_from_name(name)
+        if Group.objects.filter(name=value).exists():
+            raise serializers.ValidationError(_('Invalid group name. Name already exists as a Group.'))
```

Comment:
Maybe just "A Group with that name already exists." ?

You don't need to explicitly say that the *host* name is invalid, because the serializer will actually return the errant field like this in the API response:

![image](https://user-images.githubusercontent.com/214912/82906638-8c0c8280-9f33-11ea-90e7-5a6660a730fa.png)

Reply:
Bigger issue - this need to be filtered to the inventory.

`Group.objects.filter(name=value)` -> `Group.objects.filter(name=value, inventory=self.instance.inventory)`

If there's not a `self.instance` that could be a problem, because then it changes from being a field simple field value validation to mutual field value validation. User gave an inventory value and a name value. Does the _combination_ of those two inputs create a situation which should be rejected?

To handle those cases, logic may have to be moved from `validate_name` to `validate`, where all attributes are simultaneously available.

Reply:
Good point 👍

Reply:
Also, what are the implications for smart inventory?

*Is* this something we can even enforce reliably at that level?

Reply:
... how would smart inventory get here? You can't post a host/group to it.

Reply:
Okay, that's true - I was thinking if you changed the smart inventory filter such that it introduced a _new_ host with a collision, but smart inventories don't contain groups, so ignore me.

Reply:
@ryanpetrello @AlanCoding pushed a change, let me know your thoughts when you have a moment.